packages/java-parser/src/parser.js

```diff
@@ -148,9 +148,25 @@ class JavaParser extends Parser {
 
   // hack to turn off CST building side effects during backtracking
   // TODO: should be patched in Chevrotain
-  cstPostNonTerminal(ruleCstResult, ruleName) {
+  cstPostNonTerminal(node, ruleCstResult, ruleName) {
     if (this.isBackTracking() === false) {
-      super.cstPostNonTerminal(ruleCstResult, ruleName);
+      super.cstPostNonTerminal(node, ruleCstResult, ruleName);
+      this.addNodeOffsets(node);
+    }
+  }
+
+  addNodeOffsets(node) {
```

Comment:
This seems to only scan one level down, but it needs to be infinite levels downs to find all tokens...
I think it should be as a feature in Chevrotain see:
https://github.com/SAP/chevrotain/issues/932

Reply:
> This seems to only scan one level down, but it needs to be infinite levels downs to find all tokens...

Since we are doing the process in the cstPost, I thought we have the children nodes built before the current one, so don't we just need to check one level down ? The only case that might be missing is the root of the cst since we are calling the rule as the entrypoint ? 
If we need to scan downs to find all tokens at each node, I think the performance might be affected.

> I think it should be as a feature in Chevrotain see:
> SAP/chevrotain#932

Sorry ! I have seen that thread.
So I should let chevrotain experts do it then.

Reply:
> Since we are doing the process in the cstPost, I thought we have the children nodes built before the current one

Oops, yes you are correct 👍 

> So I should let chevrotain experts do it then.

Or contribute to chevrotain 😄

Reply:
>  Or contribute to chevrotain :smile:

I would love to, but I think by the time I get the hang of Chevrotain project the feature will already be added to it :laughing:

Reply:
I doubt it, It would have some added complexity (docs / additional perf concerns)
but the basic logic is isolated to a very small area.